helpers.js

```diff
@@ -1,58 +1,147 @@
 const childProcess = require("child_process");
-const fs = require("fs");
-const pathmodule = require("path");
-const { v4: uuidv4 } = require("uuid");
-
-async function execTerraform(cmd, args, path) {
-  if (!path) {
-    throw new Error("Parameter Working Directory is required.");
-  }
-  const terraCmd = `terraform ${cmd} ${args.join(" ")}`;
-  return new Promise((resolve, reject) => {
-    childProcess.exec(terraCmd, { cwd: path }, (error, stdout, stderr) => {
-      if (error) {
-        return reject(error);
-      }
-      if (stderr) {
-        console.log(`stderr: ${stderr}`);
-      }
-      return resolve(stdout);
-    });
-  });
+const { promisify } = require("util");
+const {
+  access,
+  lstat,
+  readFile,
+  writeFile,
+} = require("fs/promises");
+const { resolve: resolvePath } = require("path");
+const ShredFile = require("shredfile");
+
+const exec = promisify(childProcess.exec);
+
+function logToActivityLog(message) {
+  // TODO: Change console.error to console.info
+  // Right now (Kaholo v4.1.2.1) console.info
+  // does not print messages to Activity Log
+  // Jira ticket: https://kaholo.atlassian.net/browse/KAH-3636
+  console.error(message);
 }
 
-function parseVars(vars) {
-  if (!vars) { // if vars is false\undefined return empty arr
-    return [];
+async function createVariablesString({
+  varFile,
+  variables,
+  secretVariables,
+  workingDirectory,
+}) {
+  let variablesText = "";
+  if (varFile) {
+    const resolvedVarFilePath = workingDirectory
+      ? resolvePath(workingDirectory, varFile)
+      : varFile;
+    if (!await pathExists(resolvedVarFilePath)) {
+      throw new Error(`Variable File path "${resolvedVarFilePath}" does not exist!`);
+    }
+    if (!await isPathFile(resolvedVarFilePath)) {
+      throw new Error(`Variable File path "${resolvedVarFilePath}" is not a file!`);
+    }
+    const fileContent = await readFile(resolvedVarFilePath);
+    variablesText += fileContent.toString();
+    variablesText += "\n";
+  }
+  if (secretVariables) {
+    variablesText += secretVariables;
+    variablesText += "\n";
   }
-  if (typeof (vars) === "string") { // if vars is string split vars by new lines
-    return vars.split("\n").map((keyVal) => `-var="${keyVal.trim()}"`);
+  if (variables) {
+    variablesText += variables;
+    variablesText += "\n";
   }
-  if (Array.isArray(vars)) { // if vars is an array only add flags
-    return vars.map((keyVal) => `-var="${keyVal}"`);
+  if (variablesText.length) {
+    variablesText = `${variablesText.trim()}\n`;
+  }
+  return variablesText;
+}
+
+async function saveToRandomTemporaryFile(content) {
+  const fileName = generateRandomTemporaryPath();
```

Comment:
I liked the old approach with `mktemp`. It was slightly more complicated, but assured us that the file has a unique name.
However let's call the command ourselves, without any external libraries. `mktemp --tmpdir kaholo_terraform_plugin.XXX` should do the job.